tests/test_transaction_context_in_hooks.py

```diff
@@ -0,0 +1,58 @@
+from openfeature.api import (
+    set_provider,
+    set_transaction_context,
+    set_transaction_context_propagator,
+)
+from openfeature.client import OpenFeatureClient
+from openfeature.evaluation_context import EvaluationContext
+from openfeature.hook import Hook
+from openfeature.provider.no_op_provider import NoOpProvider
+from openfeature.transaction_context import ContextVarsTransactionContextPropagator
+
+
+class TransactionContextHook(Hook):
+    def __init__(self):
+        self.before_called = False
+        self.transaction_attr_value = None
+
+    def before(self, hook_context, hints):
+        self.before_called = True
+        # Check if the transaction context attribute is in the hook context
+        if "transaction_attr" in hook_context.evaluation_context.attributes:
+            self.transaction_attr_value = hook_context.evaluation_context.attributes[
+                "transaction_attr"
+            ]
+        return None
+
+
+def test_transaction_context_merged_into_hook_context():
+    """Test that transaction context is merged into the hook context's evaluation context."""
+    # Set up transaction context propagator
```

Comment:
can you remove all the inline comments here, they don't add any additional value here

Reply:
Done